packages/expo-network/ios/NetworkModule.swift

```diff
@@ -21,11 +24,26 @@ public final class NetworkModule: Module {
       return getNetworkStateAsync()
     }
 
-    OnDestroy {
+    OnStopObserving {
```

Comment:
I think we still need to cancel monitoring on `OnDestroy`. `OnStopObserving` is only called when the user removes event listeners. In other words it's not called when the module (or JS context in general) is destroyed.
Also, I would recommend using a newer version of `OnStartObserving`/`OnStopObserving` that is attached to a specific event.

```suggestion
    OnStopObserving(onNetworkStateChanged) {
```